api/src/main/java/org/openmrs/ui/framework/resource/FileResourceProvider.java

```diff
@@ -33,12 +33,7 @@
 	public File getResource(String path) {
 		File resourceFile = null;
 		if (path != null) {
-			if (new File(path).isAbsolute()) {
-				resourceFile = new File(path);
-			}
-			else {
-				resourceFile = new File(OpenmrsUtil.getApplicationDataDirectory(), path);
-			}
+			resourceFile = new File(OpenmrsUtil.getApplicationDataDirectory(), path);
```

Comment:
This is still bypassable. if the "path" is user controlled they could enter something like ../../../../../../../../../../../../etc/passwd or similar which even when appended to the application directory will result in a bypass. You will want to do a getcanonicalpath on the path before appending the app directory.

Reply:
@golfhackerdave The changes to this file aren't intended to resolve that issue, merely to remove the additional support for absolute paths. The actual fix is in the code for the ResourceFactory. This is because this class isn't used directly to load code, but indirectly via a call to the ResourceFactory.

Reply:
@mseaton @mogoodrich Could we just rename this to InitializerResourceProvider and restrict it to the `configuration` sub-directory Initializer looks for?

Reply:
Fine with me, though I think I'd lean toward calling it "ConfigurationResourceProvider"... thoughts @mseaton ?

Reply:
I was going to make the same comment.  I'm fine with it, but don't tie it to Initilalizer.  +1 to @mogoodrich suggestion of ConfigurationResourceProvider